crates/net/network/tests/it/connect.rs

```diff
@@ -80,3 +86,50 @@ async fn test_connect_with_boot_nodes() {
         dbg!(ev);
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_incoming_node_id_blacklist() {
+    reth_tracing::init_tracing();
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+
+    let reth_socket = SocketAddr::new([127, 0, 0, 1].into(), 30305);
+
+    // instantiate geth and add ourselves as a peer
+    let temp_dir = tempfile::tempdir().unwrap().into_path();
+    let geth = Geth::new().data_dir(temp_dir).disable_discovery().spawn();
+    let geth_endpoint = SocketAddr::new([127, 0, 0, 1].into(), geth.port());
+    let provider = Provider::<Http>::try_from(format!("http://{geth_endpoint}")).unwrap();
+
+    // get the peer id we should be expecting
+    let geth_peer_id: PeerId =
+        provider.node_info().await.unwrap().enr.public_key().encode_uncompressed().into();
```

Comment:
@Rjected should we have a helper for returning PeerId here so that consumers don't need to know the internals of enr encode uncompressed etc?

Reply:
Yep, will make an issue for adding enode utilities, also generally documenting and providing helper functions so consumers don't need to interact with raw keys, compressed vs uncompressed, etc. Consumers should not need to know about EC point encoding to use the p2p APIs.

Currently is a sore spot that I think could lead to api misuse and confusion, so I can put more effort on this

Reply:
added in #381

Reply:
Agreed - thanks!